retrievers/multi_vector_retriever.py

```diff
@@ -114,7 +122,6 @@ def as_runnable(self) -> RunnableSerializable:
 
         elif self.mode == MultiVectorRetrieverMode.SUMMARIZE:
             summaries = self._get_document_summaries()
-            doc_ids = [str(uuid.uuid4()) for _ in self.documents]
             summary_docs = [
                 Document(page_content=s, metadata={self.id_key: doc_ids[i]})
```

Comment:
list index out of range, if user doesn't specify list of ids

`doc_ids = [str(uuid.uuid4()) for _ in self.documents]`

seems more efficient to me than, seems unnecessary to split docs if user only wants to run with summarize

`split_docs, doc_ids = self._split_documents()`